tests/suites/test_suite_ecdsa.function

```diff
@@ -212,25 +212,39 @@ void ecdsa_prim_random( int id )
     mbedtls_mpi d, r, s;
     mbedtls_test_rnd_pseudo_info rnd_info;
     unsigned char buf[MBEDTLS_MD_MAX_SIZE];
+    int test_runs = 2;
 
     mbedtls_ecp_group_init( &grp );
     mbedtls_ecp_point_init( &Q );
     mbedtls_mpi_init( &d ); mbedtls_mpi_init( &r ); mbedtls_mpi_init( &s );
     memset( &rnd_info, 0x00, sizeof( mbedtls_test_rnd_pseudo_info ) );
     memset( buf, 0, sizeof( buf ) );
 
-    /* prepare material for signature */
-    TEST_ASSERT( mbedtls_test_rnd_pseudo_rand( &rnd_info,
-                                               buf, sizeof( buf ) ) == 0 );
-    TEST_ASSERT( mbedtls_ecp_group_load( &grp, id ) == 0 );
-    TEST_ASSERT( mbedtls_ecp_gen_keypair( &grp, &d, &Q,
-                                          &mbedtls_test_rnd_pseudo_rand,
-                                          &rnd_info ) == 0 );
-
-    TEST_ASSERT( mbedtls_ecdsa_sign( &grp, &r, &s, &d, buf, sizeof( buf ),
-                                     &mbedtls_test_rnd_pseudo_rand,
-                                     &rnd_info ) == 0 );
-    TEST_ASSERT( mbedtls_ecdsa_verify( &grp, buf, sizeof( buf ), &Q, &r, &s ) == 0 );
+    while ( test_runs-- )
+    {
+        /* prepare material for signature */
+        if ( test_runs == 1 )
+        {
+            TEST_ASSERT( mbedtls_test_rnd_pseudo_rand( &rnd_info,
+                                                       buf, sizeof( buf ) )
+                                                     == 0 );
+        } else {
+            TEST_ASSERT( mbedtls_test_rnd_zero_rand( NULL,
```

Comment:
This test function is called `ecdsa_prim_random`. Its job is to test ECDSA on random data. Its job is not to test ECDSA with a known payload. Please add test cases with all-bits-zero inputs to known-answer tests (for deterministic ECDSA) and sign-verify tests (for randomized ECDSA).